src/lib/starRelease.js

```diff
@@ -0,0 +1,86 @@
+import { linearSR, conditionalSR } from 'azimuth-js';
+import { Just, Nothing } from 'folktale/maybe';
+
+function mapRange(amount, cb) {
+  return [...Array(amount).keys()].map(cb);
+}
+
+// Preferentially withdraw from conditional contracts and use linear
+// to mop up the rest
+export function generateWithdrawTxs(contracts, batchLimits, amount, to) {
+  const [left, ...csrTxs] = generateConditionalWithdrawTxs(
+    contracts,
+    batchLimits,
+    amount,
+    to
+  );
+  const lsrTxs = generateLinearWithdrawTxs(contracts, left, to);
+  return [...csrTxs, ...lsrTxs];
+}
+
+function generateConditionalWithdrawTxs(
```

Comment:
A more explicit loop (instead of recursion) might be a bit easier to follow here. Or at least annotate that this is running itself for every batch/limit.